### PR TITLE
Bugfix: Wrong encoding method used for printer name

### PR DESCRIPTION
**Pull request by DavBfr**
Fix printing if printer name contains Unicode characters.

### CODE REVIEW

**Comment by bmiklautz** [comment]
@freerdp-bot test.

File: channels/printer/client/printer_win.c [hunk 1]
```diff
@@ -239,8 +238,7 @@ static rdpPrinter** printer_win_enum_printers(rdpPrinterDriver* driver)
 	rdpPrinter** printers;
 	int num_printers;
 	int i;
-	char pname[1000];
-	size_t charsConverted;
+	char* pname = NULL;
```
Comment (akallabeth) [inline channels/printer/client/printer_win.c]: Unused variable

File: channels/printer/client/printer_win.c [hunk 2]
```diff
@@ -206,8 +206,7 @@ static rdpPrinter* printer_win_new_printer(rdpWinPrinterDriver* win_driver,
 	win_printer->printer.FindPrintJob = printer_win_find_printjob;
 	win_printer->printer.Free = printer_win_free_printer;
 
-	swprintf(wname, 256, L"%hs", name);
-	OpenPrinter(wname, &(win_printer->hPrinter), NULL);
+	OpenPrinter(name, &(win_printer->hPrinter), NULL);
```
Comment (akallabeth) [inline channels/printer/client/printer_win.c]: Maybe add proper return value checks when already at it

File: channels/printer/client/printer_win.c [hunk 3]
```diff
@@ -194,7 +193,8 @@ static rdpPrinter* printer_win_new_printer(rdpWinPrinterDriver* win_driver,
 		return NULL;
 
 	win_printer->printer.id = win_driver->id_sequence++;
-	win_printer->printer.name = _strdup(name);
+	ConvertFromUnicode(CP_UTF8, 0, name, -1, &win_printer->printer.name, 0, NULL, NULL);
```
Comment (akallabeth) [inline channels/printer/client/printer_win.c]: Missing return value check

File: channels/printer/client/printer_win.c [hunk 4]
```diff
@@ -206,8 +206,7 @@ static rdpPrinter* printer_win_new_printer(rdpWinPrinterDriver* win_driver,
 	win_printer->printer.FindPrintJob = printer_win_find_printjob;
 	win_printer->printer.Free = printer_win_free_printer;
 
-	swprintf(wname, 256, L"%hs", name);
-	OpenPrinter(wname, &(win_printer->hPrinter), NULL);
+	OpenPrinter(name, &(win_printer->hPrinter), NULL);
 
 	GetPrinter(win_printer->hPrinter, 2, (LPBYTE) prninfo, 0, &needed);
```
Comment (akallabeth) [inline channels/printer/client/printer_win.c]: Maybe add proper return value checks when already at it

**Comment by akallabeth** [comment]
@DavBfr Could you be so kind to update this pull to address the issues pointed out?

**Comment by DavBfr** [comment]
@akallabeth I thought it was pushed ...

**Comment by akallabeth** [comment]
@freerdp-bot test
@DavBfr Ah, I know that feeling ^^

**Pull request merged**